src/pg/brc20/brc20-pg-store.ts

```diff
@@ -283,14 +341,28 @@ export class Brc20PgStore {
       limit: deploy.op.lim ?? null,
       decimals: deploy.op.dec ?? '18',
     };
-    const tickers = await this.parent.sql<{ ticker: string; address: string }[]>`
+    const insertions = await this.parent.sql<{ id: string; ticker: string; address: string }[]>`
       INSERT INTO brc20_deploys ${this.parent.sql(insert)}
       ON CONFLICT (LOWER(ticker)) DO NOTHING
+      RETURNING id, ticker, address
     `;
-    if (tickers.count)
+    if (insertions.count) {
+      // Add to event history
+      const event: DbBrc20EventInsert = {
+        operation: 'deploy',
+        inscription_id: deploy.location.inscription_id,
+        genesis_location_id: deploy.location.id,
+        brc20_deploy_id: insertions[0].id,
+        deploy_id: insertions[0].id,
+        mint_id: null,
+        transfer_id: null,
+      };
+      await this.parent.sql`INSERT INTO brc20_events ${this.parent.sql(event)}`;
```

Comment:
Same here about putting this INSERT into a CTE on the query above

Reply:
To make sure you only insert an event with a succesful deploy you could do
```sql
WITH deploy_insert AS (
  INSERT INTO brc20_deploys ${this.parent.sql(insert)}
  ON CONFLICT (LOWER(ticker)) DO NOTHING
  RETURNING id
),
INSERT INTO brc20_events (operation, inscription_id, genesis_location_id, brc20_deploy_id, deploy_id)
(SELECT 'deploy' AS operation, ${deploy.location.inscription_id} AS inscription_id, ${deploy.location.id} AS genesis_location_id, id AS brc20_deploy_id, id AS deploy_id
FROM deploy_insert)
```

Reply:
Oof this was easy yea, I didn't think about just injecting into the SELECT return as a literal 🙃